hivemq-edge/src/main/java/com/hivemq/configuration/service/impl/DataCombiningConfigurationServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2019-present HiveMQ GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hivemq.configuration.service.impl;
+
+import com.hivemq.configuration.entity.adapter.ProtocolAdapterEntity;
+import com.hivemq.configuration.entity.combining.DataCombinerEntity;
+import com.hivemq.configuration.service.DataCombiningConfigurationService;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataCombiningConfigurationServiceImpl implements DataCombiningConfigurationService {
+
+    private volatile @NotNull List<DataCombinerEntity> allConfigs = new ArrayList<>();
+
+    @Override
+    public @NotNull List<DataCombinerEntity> getAllConfigs() {
+        return allConfigs;
```

Comment:
Shouldn't that be a copy of the list rather than the full reference?

Reply:
same as ProtocolAdapterConfigurationServiceImpl. Yeah a copy or an immutable view could make sense